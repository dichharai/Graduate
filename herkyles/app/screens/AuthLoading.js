import React, {Component} from "react";
import { StyleSheet, Button, Text, View, AsyncStorage, ActivityIndicator, StatusBar } from "react-native";
import  CustomActivityIndicator  from "../components/Status/CustomActivityIndicator";



class AuthLoading extends React.Component{
  constructor(props){
    super(props);
    this._bootstrapAsync();
  }
  // Fetch the token from storage then navigate to appropriate place
  _bootstrapAsync = async () => {
    const userToken = await AsyncStorage.getItem("userToken");
    console.log(AsyncStorage.getAllKeys());
    // This will switch to the App screen or Auth screen and this loading screen will be unmounted and thrown away

    console.log("userToken: " + userToken);
    this.props.navigation.navigate(userToken? "Auth": "App");
  }
  // render any loading content that you like here
  render(){
    return (
      <CustomActivityIndicator/>
    );
  }
};

export default AuthLoading; 
  