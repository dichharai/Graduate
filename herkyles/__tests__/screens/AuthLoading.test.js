import React from "react";
import AuthLoading from "../../app/screens/AuthLoading";

import renderer from "react-test-renderer";

describe ("AuthLoading", () => {
  it("renders without crashing", () => {
    const rendered = renderer.create(<AuthLoading />).toJSON();
    expect(rendered).toBeTruthy();
    expect(rendered).toMatchSnapshot();
  });
  it('should call navigate', async () => {
    const navigationMock = {navigate: jest.fn()}; 

    let authComponent = renderer.create(<AuthLoading 
        navigation={navigationMock}
      />
    ).getInstance(); 

    authComponent._bootstrapAsync(); 
    await expect(navigationMock.navigate).toHaveBeenCalledTimes(0); 
 
  
  });
});