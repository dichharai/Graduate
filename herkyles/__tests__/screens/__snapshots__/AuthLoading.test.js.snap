// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthLoading renders without crashing 1`] = `
<View
  style={undefined}
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="small"
  />
</View>
`;
